### PR TITLE
Add optional properties to Inference node

### PR DESCRIPTION
**Pull request by SIGLOST**

### CODE REVIEW

File: pipeline/src/node_properties/model_inference_properties.rs [hunk 1]
```diff
@@ -48,5 +48,16 @@ pub struct ModelInferenceRuntime {
...
+    /// Relative difference between sides of the rectangles to merge them into a group
+    /// Used in OpenCV groupRectangles function and DBSCAN algorithm
+    pub eps: Option<f32>,
+    /// Minimum height in pixels to consider a detected object
+    pub detected_min_h: Option<i32>,
+    /// Minimum width in pixels to consider a detected object
+    pub detected_min_w: Option<i32>,
+    /// Maximum height in pixels to consider a detected object
+    pub detected_max_h: Option<i32>,
+    /// Minimum width in pixels to consider a detected object
```
Comment (DmitrySamoylov) [inline pipeline/src/node_properties/model_inference_properties.rs:61]:
```suggestion
    /// Maximum width in pixels to consider a detected object
```

**Review by DmitrySamoylov — APPROVED** [review]

File: pipeline/src/node_properties/model_inference_properties.rs [hunk 2]
```diff
@@ -48,5 +48,16 @@ pub struct ModelInferenceRuntime {
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
 pub struct ClassInferenceProperties {
     /// Minimum inference threshold, should be set on the [0.0, 1.0] interval.
-    pub min_inference_threshold: f32,
+    pub min_inference_threshold: Option<f32>,
```
Comment (jplatte) [inline pipeline/src/node_properties/model_inference_properties.rs:51]: There's no mention in the CH story or the commit message why this existing property is being made optional.

**Review by jplatte — COMMENTED** [review]
Should we be using `serde(skip_serializing_if = "Option::is_none")` here (or the attribute macro that auto-applies that to all optional fields)?

File: pipeline/src/node_properties/model_inference_properties.rs (hunk 2 above)
Comment (SIGLOST) [reply pipeline/src/node_properties/model_inference_properties.rs:51]: When we had a single inference property, it was possible to figure if it was enabled or disabled just based on the `class_properties`
```pub class_properties: Option<BTreeMap<String, ClassInferenceProperties>>,```

but now we can have the case where user sets the min/max bbox dimensions, but does not tweak the `min_inference_threshold`

Comment (DmitrySamoylov) [reply pipeline/src/node_properties/model_inference_properties.rs:51]: Yeah, this change should be in a separate commit IMO.

Comment (SIGLOST) [reply pipeline/src/node_properties/model_inference_properties.rs:51]: alternative would be remove the options and console fill the not set properties with a negative value in order we can discard it later on lumeod.

Comment (SIGLOST) [reply pipeline/src/node_properties/model_inference_properties.rs:51]: > Yeah, this change should be in a separate commit IMO.

OK, I will extract it to a separate commit and add the reasoning on commit log as soon we settle the changes on the main commit.

File: pipeline/src/node_properties/model_inference_properties.rs [hunk 3]
```diff
@@ -45,8 +47,16 @@ pub struct ModelInferenceRuntime {
...
 }
 
+#[skip_serializing_none]
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
 pub struct ClassInferenceProperties {
     /// Minimum inference threshold, should be set on the [0.0, 1.0] interval.
-    pub min_inference_threshold: f32,
+    pub min_inference_threshold: Option<f32>,
+    /// Relative difference between sides of the rectangles to merge them into a group
+    /// Used in OpenCV groupRectangles function and DBSCAN algorithm
```
Comment (jplatte) [inline pipeline/src/node_properties/model_inference_properties.rs:56]: Tiny nit: When the docs consist of multiple sentences the first one should be a "summary" line and be on its own paragraph.

```suggestion
    /// Relative difference between sides of the rectangles to merge them into a group
    ///
    /// Used in OpenCV groupRectangles function and DBSCAN algorithm
```

(and maybe add periods at the end of the sentences everywhere)

**Review by jplatte — APPROVED** [review]

**Pull request merged**